### ISSUE TITLE
Getting borged by malf AI robotic factory after AI death can result in law weirdness

### ISSUE THREAD

**Issue by Angustmeta** [tgstation/tgstation#43743]
#### Round ID:
107069 Terry

#### Testmerges:
#43682
#43148

#### Reproduction:
I went to the AI sat as geneticist to help kill the malf AI. There the sat was already semi-blown to bits and I met a Durand who's pilot said the AI was spaced. So I assume the AI was dead already at that point. 

I went towards departures (Metastation) and saw a robo factory in xenobio, next to some semingly docile or broken borgs and the CMO. Any moving borg I saw didn't seem to attack the CMO, but CMO was being eaten by a slime. 
[Screenshot](https://i.imgur.com/r8Zp8nx.jpg) 

CMO had been nice that round so I ran in to try and save him, but got stunned by one borg, and chain-flashed by another and dragged to the robo factory. After a cooldown I was borged and met with this message: 
[Screenshot](https://i.imgur.com/MdRy0EH.jpg)

Basically unsynched Asimov. This felt weird but I went "Okay, I guess that's how it works if the malf AI is dead". So I saved a human and dragged it to departures, removed shocks and bolts from doors that I was passing. Something felt weird though, so I checked laws under Robot Commands (I only have 2 hours as cyborg), and there I had the law 0, and I believe it said I did not have an AI. 

Then on the ending screen (still alive) I was not on the list of cyborgs at all:
[Screenshot](https://i.imgur.com/hPQN2LB.jpg)
My cyborg name was "Classic Shell", and ckey is "Angust". 

Am I right to assume the laws you see when borged are supposed to be your actual laws? Then this is a pretty major bug imo. 

The only similar issue I found is closed and old as fuck: #23234

**Comment by YakumoChen** [comment]
Probably intended to avoid any shittiness with malf AIs and not having borgs synced to you. Autoborgers from a malf AI should make you a malf borg, AI or not, makes sense to me.

**Comment by AlinaStarkova** [comment]
this isnt a bug with the borging or laws but rather the display, back in 2019 and early 2020 i believe, when a borg "spawned" (such as round start) it always assumed and displayed asimov laws, law 0 was not actually displayed in chat unless you specifically hit the show laws verb to "refreash" the laws, i as a borg player at this time even had to make it a habit to hit the show laws verb as a borg everytime a new round started to ensure im not missing my malf status.

IIRC this bug has been fixed as everytime im a borg to a malf AI these days my law 0 always properly displays so this issue may be able to be closed

**Comment by timothymtorres** [comment]
So this appears to be an edge case issue with the malf robo factory.  Since the malf AI is dead, any borgs that are created using the robo factory are unsynched to other AIs and don't get malf laws.  Honestly, the robo factory should just explode when a malf AI is killed but for now I think this behavior is fine the way it is and I wouldn't consider this outcome to be a bug.    

I'm linking #44349 and #25703 since they are somewhat related although they aren't duplicates.

**Comment by AlinaStarkova** [comment]
> 
> 
> So this appears to be an edge case issue with the malf robo factory. Since the malf AI is dead, any borgs that are created using the robo factory are unsynched to other AIs and don't get malf laws. Honestly, the robo factory should just explode when a malf AI is killed but for now I think this behavior is fine the way it is and I wouldn't consider this outcome to be a bug.
> 
> I'm linking #44349 and #25703 since they are somewhat related although they aren't duplicates.

i just explained above why this isnt a bug anymore, you DO get malf laws, but before we had tgui chat the law 0 wouldnt show up when you became a borg even if you had it, hence the confusion, that issue no longer exists so this now works as intended
